libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CoreBotAdapter.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    /// <summary>
+    /// CoreBotAdapter is a base BotAdapter that derives from CloudAdapter and implements OnTurnError handling. This class
+    /// also registers any middleware services that are contained in the middlewares collection.
+    /// </summary>
+    public class CoreBotAdapter : CloudAdapter
```

Comment:
Does this class require new public surface area? Could it just be hidden in this assembly?

Reply:
I wasn't sure if we could make it internal so I gave it a try but that doesn't work since the DI system can't create an instance since its external from this assembly. Without CoreBotAdapter there would be no way to set up a very basic bot (typically in a sample or demo) without requiring the user to provide their own implementation since we can't infer what their implementation is unless they declare it. While this adds to the SDK, it removes a redundant implementation from every sample unless they need a special implementation for some reason. We could not provide it and remove the basic AddBotRuntime method and require the user to provide their IBot implementation type. I'm open on which way we go here.

Reply:
Given that the registration is in the Extensions class that you are defining in this assembly, we should be able to make this internal. Happy to debug if you run into issues but in general it's a good thing to keep it internal to save ourselves a lot of the past pain for making everything public by default.

Reply:
Got it, I had absent mindedly set the constructor to internal as well as the class, which is why the DI container couldn't create it. I've corrected this and will push an update.